### PR TITLE
fix(local): fetch missing metadata from acousticId api

### PR DESCRIPTION
**Pull request by charjac**
for now its just a POC. it look like it works on linux but i don't have enough mp3 to be sure neither a windows and mac computer for test (too lazy for a vm :)).

### CODE REVIEW

**Comment by nukeop** [comment]
I might try to test it on a vm today.

**Comment by nukeop** [comment]
I got around to this finally. It doesn't build for me on Windows, unfortunately.

**Comment by charjac** [comment]
:'( :'( :'(
Could you copy paste the error log here please ?

I will fight my laziness and try to fix it in a vm \0/.

**Comment by nukeop** [comment]
i will try to reproduce it, but i didn't paste it because it wasn't very descriptive. maybe i'll try running it natively.

**Comment by charjac** [comment]
I try to build nuclear on a windows10 64 family edition, the build was ok, but it fails at running :'(.
Here is the error:

Capture d’écran de 2019-07-01 13-46-02

**Comment by charjac** [comment]
In development i got this:

Capture d’écran de 2019-07-01 16-20-53

looks like the error is triggered by lodash in electron-store package .... 😰😰😰

**Comment by charjac** [comment]
i get the same error when trying to build from master :/

Capture d’écran de 2019-07-01 17-56-45

**Comment by charjac** [comment]
but when i download the portable .exe from release tab it works .......
windows is so fucking drunk :'( :'( :'( 

How and where do you build it ?

**Comment by nukeop** [comment]
I manually patched some stuff in the compiled javascript to get rid of this error, it's not viable long term.
This great contributor patched the third party package that was the root of this issue (I think): https://github.com/nukeop/nuclear/issues/362#issuecomment-506302785
I'll update package.json.
At least I think that was the related issue.

**Comment by charjac** [comment]
if the package swagger-spec-express is a problem, i can easily replace it with something else.

**Comment by nukeop** [comment]
I rebuilt with the patched version of the package and it indeed looks like the issue is gone.

**Comment by nukeop** [comment]
Another person is contributing an upgrade for the travis config file so that we can have continuously built Windows binaries too: https://github.com/nukeop/nuclear/pull/380

When that's merged we can move to releasing on every push.

**Comment by charjac** [comment]
sounds great !!!
Could you push the new package.json to master so i can rebase and continue my investigation ?

**Comment by nukeop** [comment]
Ok, done.

**Comment by charjac** [comment]
thx

**Comment by charjac** [comment]
finally i found the problem:
it come from app/persistence/store.js

it seems like electron-store doesn't like to set a key with an undefined value on windows.....

it still dont run with wine (not really surprised) but has a new error \0/, i try on a vm later today.

**Comment by charjac** [comment]
I think we are good now.

**Comment by charjac** [comment]
for now the acousticId licence is hard coded and associated with my email address.
I think you should create your own and then we should think about where we put it (environment ? settings ?).

**Comment by nukeop** [comment]
Normally I put all the keys in `globals.js`. I can register one on nuclear@gumblert.tech, which serves as the official email address for Nuclear.

**Comment by charjac** [comment]
This key is intended to be used in the node.js process so for now we could leave it here.

I was thinking about a great refactor of the server part.
With all the recent new feature, it become a little messy.
What do you think ?

**Comment by nukeop** [comment]
Yeah, definitely. I'll need to unify the server file so that it detects the environment (dev/prod) and platform and we don't have to keep the files in sync. Other than that i think the modules are separated pretty well but if you have ideas how to improve that further, feel free to implement them.

**Comment by charjac** [comment]
ok cool as soon as it works normally on windows i go on this.

still buggy on windows ...

**Comment by charjac** [comment]
looks like its ok now

**Comment by nukeop** [comment]
Great, I'm gonna check it out soon, and merge if it works fine.

**Comment by charjac** [comment]
Looks like the rebase has broke something .... I check this later today.

**Comment by nukeop** [comment]
I refactored that bit of the code that was broken earlier. If this broke your changes I'll try to fix it myself.

**Comment by charjac** [comment]
ok its nothing.
Actually the library view can only works if api is enabled.

We can:
  - add a message in the view if api is not enabled
  - separate the part of the api wich is needed for the library to work

**Comment by nukeop** [comment]
I think 1) is more realistic.

**Comment by charjac** [comment]
ready for review :)

File: package.json [hunk 1]
```diff
@@ -69,9 +71,12 @@
     "moment": "^2.20.1",
     "mousetrap": "^1.6.2",
     "music-metadata": "^3.5.3",
+    "node-fetch": "^2.6.0",
```
Comment (nukeop) [inline package.json]: You don't need that, we have isomorphic-fetch already.

File: package.json [hunk 2]
```diff
@@ -69,9 +71,12 @@
...
     "nuclear-core": "0.0.5",
     "nuclear-ui": "0.0.6",
     "numeral": "^2.0.6",
+    "once": "^1.4.0",
```
Comment (nukeop) [inline package.json]: Is this used anywhere? In any case it's better to use lodash's `_.once`

File: server/lib/platform.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { platform } from 'os';
```
Comment (nukeop) [inline server/lib/platform.js]: You don't need that, we already have `electron-platform` (https://www.npmjs.com/package/electron-platform). And it has all the platform detection functions you need. I think this also gets rid of the `os` dependency.

**Comment by nukeop** [comment]
Looks good, I wonder how this work in practice. How fast is it? Maybe it'd be worthwhile to provide an option to skip all unidentified tracks instead of looking up their metadata?

File: package.json (hunk 2 above)
Comment (charjac) [reply package.json]: mmmh i think it was stream related, and i finally remove it during a refacto

**Comment by charjac** [comment]
I dont know, Actually the meta are cached in the store, so if the user doesn't move his files the meta are only fetched once (if no meta present in the file).

**Comment by charjac** [comment]
thx ! I start working on the server refactor

**Pull request merged**